lxd/db/cluster/storage_volume_type.go

```diff
@@ -19,12 +31,50 @@ const (
 	StoragePoolVolumeTypeNameCustom    string = "custom"
 )
 
-// StoragePoolVolumeTypeNames represents a map of storage volume types and their names.
-var StoragePoolVolumeTypeNames = map[int]string{
-	StoragePoolVolumeTypeContainer: "container",
-	StoragePoolVolumeTypeImage:     "image",
-	StoragePoolVolumeTypeCustom:    "custom",
-	StoragePoolVolumeTypeVM:        "virtual-machine",
+// StoragePoolVolumeTypeFromInt is a checked conversion to StoragePoolVolumeType.
+func StoragePoolVolumeTypeFromInt(volType int) (StoragePoolVolumeType, error) {
+	switch StoragePoolVolumeType(volType) {
+	case StoragePoolVolumeTypeContainer, StoragePoolVolumeTypeVM, StoragePoolVolumeTypeCustom, StoragePoolVolumeTypeImage:
+		return StoragePoolVolumeType(volType), nil
+	default:
+		return StoragePoolVolumeType(volType), errors.New("Invalid storage volume type")
+	}
+}
+
+// StoragePoolVolumeTypeFromName is a checked conversion to StoragePoolVolumeType.
+func StoragePoolVolumeTypeFromName(volTypeName string) (StoragePoolVolumeType, error) {
+	switch volTypeName {
+	case StoragePoolVolumeTypeNameContainer:
+		return StoragePoolVolumeTypeContainer, nil
+	case StoragePoolVolumeTypeNameVM:
+		return StoragePoolVolumeTypeVM, nil
+	case StoragePoolVolumeTypeNameImage:
+		return StoragePoolVolumeTypeImage, nil
+	case StoragePoolVolumeTypeNameCustom:
+		return StoragePoolVolumeTypeCustom, nil
+	}
+
+	return StoragePoolVolumeTypeCustom, errors.New("Invalid storage volume type")
+}
+
+// Name gives the name of a StoragePoolVolumeType.
+//
+// # Safety
+// This function assumes that `t` is one of the StoragePoolVolumeType enums
+// defined above.
+func (t StoragePoolVolumeType) Name() string {
```

Comment:
Maybe String() instead of Name() so it aligns with Go's conventions?

Reply:
Resolved by https://github.com/canonical/lxd/pull/15175